src/backend/utils/misc/guc.c

```diff
@@ -8943,7 +8944,9 @@ void
 ExecSetVariableStmt(VariableSetStmt *stmt, bool isTopLevel)
 {
 	GucAction	action = stmt->is_local ? GUC_ACTION_LOCAL : GUC_ACTION_SET;
+	bool		need_dispatch = false;
 
+	need_dispatch = CdbDoesNeedDispatchCommand(stmt);
```

Comment:
I think we should use hook 'CdbDoesNeedDispatchCommand'' here instead of 'CdbDoesNeedDispatchCommand', and add set dispatch logic in 'CdbDoesNeedDispatchCommand'.